clients/src/main/java/org/apache/kafka/clients/producer/Partitioner.java

```diff
@@ -44,4 +44,14 @@
      */
     public void close();
 
+
+    /**
+     * Executes right before a new batch will be created. For example, if a sticky partitioner is used,
```

Comment:
Maybe "notifies the partitioner that a new batch is about to be created... when using the sticky partitioner..."